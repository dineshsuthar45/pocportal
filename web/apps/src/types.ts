import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
