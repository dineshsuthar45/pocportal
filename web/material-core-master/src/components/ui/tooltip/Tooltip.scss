.m-tooltip-root {
	&.tooltip-container {
		overflow: hidden;
		text-overflow: ellipsis;
		white-space: nowrap;
	}
}
