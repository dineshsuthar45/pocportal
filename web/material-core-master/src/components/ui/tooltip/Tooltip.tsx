import MuiTooltip from "@mui/material/Tooltip";
import { useCallback, useEffect, useRef, useState } from "react";
import { TooltipProps } from "./Tooltip.types";

import "./Tooltip.scss";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/tooltip/
 */
export const MTooltip = ({
  children,
  title,
  arrow = true,
  showAlways = false,
  placement = "bottom",
  ...props
}: TooltipProps) => {
  const tooltipContainerRef = useRef<HTMLDivElement>(null);
  const [tooltipTitle, setTooltipTitle] = useState(title);
  const [showTooltip, setShowtooltip] = useState(false);
  const [hasOverflow, setHasOverflow] = useState(false);
  const [hoverStatus, setHover] = useState(false);

  const checkOverflown = useCallback(() => {
    if (tooltipContainerRef.current) {
      setHasOverflow(
        tooltipContainerRef.current.scrollWidth >
          tooltipContainerRef.current.clientWidth
      );
    }
  }, [
    tooltipContainerRef.current,
    tooltipContainerRef.current?.scrollWidth,
    tooltipContainerRef.current?.clientWidth,
  ]);

  useEffect(() => {
    setShowtooltip(showAlways || hasOverflow);
  }, [showAlways, hasOverflow]);

  useEffect(() => {
    if (showTooltip) {
      setTooltipTitle(title);
    } else {
      setTooltipTitle("");
    }
  }, [showTooltip, title]);

  return (
    <MuiTooltip
      title={tooltipTitle}
      open={hoverStatus && (hasOverflow || showAlways)}
      arrow={arrow}
      placement={placement}
      {...props}
      className={`m-tooltip-root ${props.className ?? ""}`.trim()}
      onMouseEnter={(e) => {
        setHover(true);
        checkOverflown();
        if (props.onMouseEnter) {
          props.onMouseEnter(e);
        }
      }}
      onMouseLeave={(e) => {
        setHover(false);
        if (props.onMouseLeave) {
          props.onMouseLeave(e);
        }
      }}
    >
      <div ref={tooltipContainerRef} className="tooltip-container">
        {children}
      </div>
    </MuiTooltip>
  );
};

MTooltip.defaultProps = {
  arrow: true,
  placement: "bottom",
  showAlways: false,
};
