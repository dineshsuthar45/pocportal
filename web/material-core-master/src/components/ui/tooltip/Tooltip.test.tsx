import { render } from "@testing-library/react";
import React from "react";
import { MTooltip } from "./Tooltip";
import { TooltipProps } from "./Tooltip.types";

describe("Tooltip component", () => {
  const HelloWorld = React.forwardRef<HTMLButtonElement>((props, ref) => (
    <button ref={ref} {...props}>
      Hello World!
    </button>
  ));
  const renderComponent = () => {
    return render(<MTooltip {...props} />);
  };
  let props: TooltipProps;

  beforeEach(() => {
    props = {
      title: "Some tooltip text here",
      children: <HelloWorld />,
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it("should render with tooltip correctly", () => {
    props.showAlways = true;
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
