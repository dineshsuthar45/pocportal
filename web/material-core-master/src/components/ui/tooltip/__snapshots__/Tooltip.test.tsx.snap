// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component should render correctly 1`] = `
<div>
  <div
    aria-label=""
    class="m-tooltip-root tooltip-container"
    data-mui-internal-clone-element="true"
  >
    <button>
      Hello World!
    </button>
  </div>
</div>
`;

exports[`Tooltip component should render with tooltip correctly 1`] = `
<div>
  <div
    aria-label="Some tooltip text here"
    class="m-tooltip-root tooltip-container"
    data-mui-internal-clone-element="true"
  >
    <button>
      Hello World!
    </button>
  </div>
</div>
`;
